public/js/Lineplot.js

```diff
@@ -209,56 +216,90 @@ Lineplot.prototype.updateActiveLine = function(feature) {
+        this._handleFeatureChange();
     }
 };
 
 /**
- * keypressUpdate: Update active feature line on keypress.
+ * onKeydownUpdate: Update active feature line on keypress.
  *
  * On pressing the left or right arrow key, the active feature
- * line and feature histogram is updated.
+ * is updated. This function checks that the change is valid
+ * and handles the update.
  *
  * @this {Lineplot}
  * @param {number} keyCode - The Javascript character code for the
  *     pressed key.
  *     See https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode
+ * @private
  */
-Lineplot.prototype.keypressUpdate = function(keyCode) {
-    var self = this;
-
-    if(self.activeFeature) {
-        if(keyCode === RIGHTARROW &&
-            // only let the line move right if it's not already at the
-            // rightmost feature
-            self.activeFeature < this.featureVector.length) {
-            self.activeFeature++;
-            $('body').trigger('updateLineplot', self.activeFeature);
-        }
-        else if(keyCode === LEFTARROW && self.activeFeature > 1) {
-            // only let the line move left if it's not already at the
-            // left-most feature
-            self.activeFeature--;
-            $('body').trigger('updateLineplot', self.activeFeature);
-        }
+Lineplot.prototype._onKeydownUpdate = function(keyCode) {
+    // only let the line move right if it's not already at the
+    // rightmost feature
+    if(keyCode === RIGHTARROW &&
+        this.activeFeature < this.featureVector.length) {
+        this.activeFeature++;
+        this._handleFeatureChange();
+    }
+    // only let the line move left if it's not already at the
+    // left-most feature
+    else if(keyCode === LEFTARROW && this.activeFeature > 1) {
+        this.activeFeature--;
+        this._handleFeatureChange();
     }
 };
 
 /**
- * destroy: Remove all child SVG elements of the plot objects containing div.
+ * setScale: Set scale for current feature vector.
+ *
+ * Sets the scale that maps the data to the plot.
+ *
+ * @private
+ */
+Lineplot.prototype._setScale = function() {
+    // get min/max x/y values -- needed for scaling axis/data
+    var yMin = d3.min(this.featureVector);
+    var yMax = d3.max(this.featureVector);
+
+    // create margin for y axis -- stops the line from
+    // drawing at the very top or bottom of the plot
+    var yMargin = (yMax - yMin) * this.axisMargin;
+
+    this.xScale = d3.scale.linear()
+        .domain([0, this.featureVector.length])
+        .range([0, this.width]);
+
+    this.yScale = d3.scale.linear()
+        .domain([yMin - yMargin, yMax + yMargin])
+        .range([this.height, 0]);
+};
+
+/**
+ * updateActiveLine: Redraw the line overlaying selected feature in the plot.
  *
  * @this {Lineplot}
+ * @private
  */
-Lineplot.prototype.destroy = function() {
-    d3.select(this.element + ' > svg').remove();
+Lineplot.prototype._updateActiveLine = function() {
```

Comment:
"Lineplot" is a very generic name, whereas this function is for a very specific line in a very specific plot. So there's some API tweaking to be done here...